src/backend/distributed/utils/metadata_cache.c

```diff
@@ -3089,7 +3097,11 @@ ResetDistTableCacheEntry(DistTableCacheEntry *cacheEntry)
 static void
 InvalidateForeignRelationGraphCacheCallback(Datum argument, Oid relationId)
 {
-	/* when invalidation happens simply set the LocalGroupId to the default value */
+	/*
+	 * We shouldn't call DistColocationRelationId() here since cache
+	 * invalidation can happen before postgres' relation cache being
+	 * initialized, in which case get_relname_relid() fails.
```

Comment:
this comment is not accurate. I'll update it after spending time on understanding things.